### PR TITLE
[Filebeat] Fix id for config map 

### PR DESCRIPTION
**Pull request by constanca-m**
This PR is a fix to a bug introduced in https://github.com/elastic/beats/pull/37401.

Defining the `id` as

```yaml
id: kubernetes-container-logs-${data.kubernetes.pod.name}-${data.kubernetes.container.id}
```

Causes the error:

```logs
{"log.level":"error","@timestamp":"2024-01-04T13:13:02.933Z","log.origin":{"file.name":"instance/beat.go","file.line":1307},"message":"Exiting: failed to read input configuration: failed to unpack input configuration: missing field accessing 'filebeat.inputs.0.id' (source:'/etc/filebeat.yml')","service.name":"filebeat","ecs.version":"1.6.0"}
Exiting: failed to read input configuration: failed to unpack input configuration: missing field accessing 'filebeat.inputs.0.id' (source:'/etc/filebeat.yml')
```

We need to remove the fields `${data.*}` in the `id` to fix it.

### CODE REVIEW

**Comment by gsantoro** [comment]
Do you mind updating also:
- https://github.com/elastic/beats/blob/c9f1426f12d9f9cdfb317c747118e2744abdccd7/dev-tools/kubernetes/filebeat/manifest.run.yaml#L116
- https://github.com/elastic/beats/blob/c9f1426f12d9f9cdfb317c747118e2744abdccd7/dev-tools/kubernetes/filebeat/manifest.debug.yaml#L116

to keep dev-tools manifests up to date with deploy manifests

File: deploy/kubernetes/filebeat-kubernetes.yaml [hunk 1]
```diff
@@ -113,14 +113,15 @@ data:
   filebeat.yml: |-
     filebeat.inputs:
     - type: filestream
-      id: kubernetes-container-logs-${data.kubernetes.pod.name}-${data.kubernetes.container.id}
+      id: kubernetes-container-logs
```
Comment (rdner) [inline deploy/kubernetes/filebeat-kubernetes.yaml:116]: These substitutions should work only in autodiscover, not in regular input configuration, AFAIK.

I see @gsantoro made this change in https://github.com/elastic/beats/pull/37401
How was it tested?

**Review by rdner — CHANGES_REQUESTED** [review]

File: deploy/kubernetes/filebeat-kubernetes.yaml [hunk 2]
```diff
@@ -113,14 +113,15 @@ data:
...
       paths:
         - /var/log/containers/*.log
       parsers:
         - container: ~
       prospector:
         scanner:
           fingerprint.enabled: true
+          fingerprint.length: 300
```
[the remark this answers is not in the document]
Comment (constanca-m) [reply deploy/kubernetes/filebeat-kubernetes.yaml:124]: Thank you for the explanation @rdner , so the warning does not cause unexpected behavior? I will remove that line in that case.

[the remark this answers is not in the document]
Comment (rdner) [reply deploy/kubernetes/filebeat-kubernetes.yaml]: @constanca-m The warning is just to notify the user why the ingestion of the file got delayed. Perhaps we could re-phrase this warning to be more clear and less error-looking. Ideas are welcome.

It does not cause any incorrect/unexpected behaviour, it's a normal state of things.

File: deploy/kubernetes/filebeat-kubernetes.yaml (hunk 1 above)
Comment (rdner) [reply deploy/kubernetes/filebeat-kubernetes.yaml:116]: @constanca-m My point is rather if this was not tested, how can we trust the rest of the changes from that PR? Perhaps we need to ask @gsantoro to re-test those changes.

Comment (gsantoro) [reply deploy/kubernetes/filebeat-kubernetes.yaml:116]: hello @rdner , sorry that was my fault. It was an unintended change. I'll do some testing now anyway

**Review by rdner — APPROVED** [review]
LGTM in the current state.

**Review by gsantoro — APPROVED** [review]

File: deploy/kubernetes/filebeat-kubernetes.yaml (hunk 1 above)
Comment (gsantoro) [reply deploy/kubernetes/filebeat-kubernetes.yaml:116]: I have now tested this. it works as expected with this PR changes. thanks @constanca-m for catching this

**Pull request merged**